modules/programs/anki.nix

```diff
@@ -0,0 +1,567 @@
+      }[hide_top_bar_mode_str]
+      profile_manager.set_top_bar_hide_mode(hide_mode)
+
+    hide_bottom_bar_str: str = "${toString cfg.hideBottomBar}"
+    if hide_bottom_bar_str:
+      profile_manager.set_hide_bottom_bar(bool(hide_bottom_bar_str))
+
+    hide_bottom_bar_mode_str: str = "${toString cfg.hideBottomBarMode}"
+    if hide_bottom_bar_mode_str:
+      hide_mode: HideMode = {
+        "fullscreen": HideMode.FULLSCREEN,
+        "always": HideMode.ALWAYS,
+      }[hide_bottom_bar_mode_str]
+      profile_manager.set_bottom_bar_hide_mode(hide_mode)
+
+    reduce_motion_str: str = "${toString cfg.reduceMotion}"
+    if reduce_motion_str:
+      profile_manager.set_reduce_motion(bool(reduce_motion_str))
+
+    minimalist_mode_str: str = "${toString cfg.minimalistMode}"
+    if minimalist_mode_str:
+      profile_manager.set_minimalist_mode(bool(minimalist_mode_str))
+
+    spacebar_rates_card_str: str = "${toString cfg.spacebarRatesCard}"
+    if spacebar_rates_card_str:
+      profile_manager.set_spacebar_rates_card(bool(spacebar_rates_card_str))
+
+    legacy_import_export_str: str = "${toString cfg.legacyImportExport}"
+    if legacy_import_export_str:
+      profile_manager.set_legacy_import_export(bool(legacy_import_export_str))
+
+    answer_keys: tuple[tuple[int, str], ...] = (${
+      lib.strings.concatMapStringsSep ", " (val: "(${toString val.ease}, '${val.key}')") cfg.answerKeys
+    })
+    for ease, key in answer_keys:
+      profile_manager.set_answer_key(ease, key)
+
+    # Profile specific options
+
+    profile_manager.create("User 1")
+    profile_manager.openProfile("User 1")
+
+    # Without this, the collection DB won't get automatically optimized.
+    profile_manager.profile["lastOptimize"] = None
+
+    auto_sync_str: str = "${toString cfg.sync.autoSync}"
+    if auto_sync_str:
+      profile_manager.profile["autoSync"] = bool(auto_sync_str)
+
+    sync_media_str: str = "${toString cfg.sync.syncMedia}"
+    if sync_media_str:
+      profile_manager.profile["syncMedia"] = bool(sync_media_str)
+
+    media_sync_minutes_str: str = "${toString cfg.sync.autoSyncMediaMinutes}"
+    if media_sync_minutes_str:
+      profile_manager.set_periodic_sync_media_minutes = int(media_sync_minutes_str)
+
+    network_timeout_str: str = "${toString cfg.sync.networkTimeout}"
+    if network_timeout_str:
+      profile_manager.set_network_timeout = int(network_timeout_str)
+
+    profile_manager.save()
+  '';
+  ankiConfig =
+    let
+      cfgAnkiPython = (
+        lib.lists.findSingle (x: x.isPy3 or false) null null (cfg.package.nativeBuildInputs or [ ])
+      );
+      ankiPackage = if cfgAnkiPython == null then pkgs.anki else cfg.package;
+      ankiPython = if cfgAnkiPython == null then pkgs.python3 else cfgAnkiPython;
+    in
+    pkgs.runCommand "ankiConfig"
+      {
+        nativeBuildInputs = [ ankiPackage ];
+      }
+      ''
+        ${ankiPython.interpreter} ${buildAnkiConfig} $out
+      '';
```

Comment:
Or doe the module always require a package since we need the anki package to generate the db file?

Reply:
See the other answer, but also yest to this.